src/rgw/rgw_sal_motr.cc

```diff
@@ -1246,8 +1246,35 @@ MotrObject::~MotrObject() {
 
 int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
 {
-  // TODO: implement
-  ldpp_dout(dpp, 20) <<__func__<< ": MotrObject::set_obj_attrs()" << dendl;
+  if (this->category == RGWObjCategory::MultiMeta)
```

Comment:
why this is needed btw?

Reply:
Checks if category is multipart upload

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
> Other API operations that support tagging
[PUT Object](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html) and [Initiate Multipart Upload](https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadInitiate.html)– You can specify tags when you create objects. You specify tags using the x-amz-tagging request header.

[Initiate Multipart Upload] API supports tagging, then why this code is disabled am not sure. Still we can address this separately through planned ticket in coming sprint 70, as per[ feature tracker ](https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/968491062/F-28B+Object+Tagging+API)
"Add object tags with CreateMultipartUpload"